src/oscar/apps/dashboard/catalogue/forms.py

```diff
@@ -27,17 +29,31 @@
 
 BaseCategoryForm = movenodeform_factory(
     Category,
-    fields=['name', 'slug', 'description', 'image', 'is_public'],
+    fields=['name', 'slug', 'description', 'meta_title', 'meta_description', 'image', 'is_public'],
     exclude=['ancestors_are_public'])
 
 
-class CategoryForm(BaseCategoryForm):
+class _SEOFormMixin(object):
+    def visible_fields(self):
+        return [field for field in self if not field.is_hidden and not self.is_seo_field(field)]
+
+    def seo_fields(self):
+        return [field for field in self if self.is_seo_field(field)]
+
+    @staticmethod
+    def is_seo_field(field):
+        return re.match(r'^meta[\w_]+|slug$', field.name)
+
+
+class CategoryForm(_SEOFormMixin, BaseCategoryForm):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         if 'slug' in self.fields:
             self.fields['slug'].required = False
             self.fields['slug'].help_text = _('Leave blank to generate from category name')
+        if 'meta_description' in self.fields:
```

Comment:
Don't think we need to do this at init - can do it when defining the field itself, i.e., 

```
meta_description = models.TextField(_('Meta Description'), blank=True, widget=forms.TextArea(attrs={'class': 'no-widget-init'})
```